relayer/src/services/relayer/optimism.ts

```diff
@@ -99,11 +99,14 @@ export class OptimismRelayerService extends BaseRelayerService {
     if (status !== MessageStatus.READY_FOR_RELAY) {
       throw new Error("Message not ready for relaying.");
     }
-
     // Now we can relay the message to L1.
     console.log("Relaying...");
     try {
-      await this.messenger.finalizeMessage(message);
+      const generalGasLimit = await this.messenger.estimateGas.finalizeMessage(message);
+
+      await this.messenger.finalizeMessage(message, {
+        overrides: { gasLimit: generalGasLimit.mul(2) },
```

Comment:
Per the other discussion, this should be hardcoded to `2_000_000`, and avoid gas estimation entirely.